tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -63,8 +65,12 @@ syntaxDatatype language allSymbols datatype = skipDefined $ do
   case datatype of
     SumType (DatatypeName _) _ subtypes -> do
       types' <- fieldTypesToNestedSum subtypes
-      con <- normalC name [TH.bangType strictness (pure types' `appT` varT typeParameterName)]
-      pure [NewtypeD [] name [PlainTV typeParameterName] Nothing con [deriveGN, deriveStockClause, deriveAnyClassClause]]
+      let fieldName = mkName ("get" <> nameStr)
+      con <- recC name [TH.varBangType fieldName (TH.bangType strictness (pure types' `appT` varT typeParameterName))]
```

Comment:
This also tweaks the definitions of sums a little, from this:

```haskell
newtype Foo a = Foo ((…) a)
```

to this:

```haskell
newtype Foo a = Foo { getFoo :: (…) a }
```

This was done because there’s no easy way to splice in pattern matches in quasiquotes, so the eliminator was much tidier, but also because it’s just plain handy to have sometimes.

Reply:
I like `get` over `un`. 👍